### ISSUE TITLE
Grid footer template adds invalid aria-owns id's

### ISSUE THREAD

**Issue by yanmariomenev** [telerik/kendo-angular#4536]
### Description

From version [13.2.1](https://www.telerik.com/kendo-angular-ui/components/changelogs/kendo-angular-ui#v13.2.1), the Grid adds invalid `aria-owns` id's that target non-existing element id's which are meant to target locked or spanned columns.

### Steps To Reproduce

Open our online example and run the Axe accessibility tool or inspect the footer of the Grid.

https://demos.telerik.com/kendo-angular-ui/demos/grid/configuration/footer-template?theme=default-ocean-blue-a11y
https://www.telerik.com/kendo-angular-ui/components/grid/columns/templates#footer-template

### Screenshots or video

![Image](https://github.com/user-attachments/assets/07ecb2bd-0220-4ccb-9dad-8e9b9da84b0b)

### Expected Behavior
`aria-owns` should be added only when needed, such as in the TreeList component.

### Browser

Chrome

### Browser version

latest
